rust/src/protocol_types/governance/voting_procedure.rs

```diff
@@ -0,0 +1,65 @@
+use crate::*;
+
+#[derive(
+    Clone,
+    Debug,
+    Hash,
+    Eq,
+    Ord,
+    PartialEq,
+    PartialOrd,
+    serde::Serialize,
+    serde::Deserialize,
+    JsonSchema,
+)]
+#[wasm_bindgen]
+pub enum VoteKind {
+    No = 0,
+    Yes = 1,
+    Abstain = 2,
+}
+
+#[derive(
+    Clone,
+    Debug,
+    Hash,
+    Eq,
+    Ord,
+    PartialEq,
+    PartialOrd,
+    serde::Serialize,
+    serde::Deserialize,
+    JsonSchema,
+)]
+#[wasm_bindgen]
+pub struct VotingProcedure {
+    pub(crate) vote: VoteKind,
+    pub(crate) anchor: Option<Anchor>,
+}
+
```

Comment:
The API of `VotingProcedure::vote: VoteKind` is inconsistent with other numeric enums like `Redeemer::tag: RedeemerTag`.

In this case, we have:
```rust
enum FooKind { Variant1, Variant2 }

struct Bar {
    foo: FooKind
}
```

Elsewhere, we have:
```rust
enum FooKind { Variant1, Variant2 }

impl Foo {
    fn new_variant1() -> Self {
        Foo(FooKind::Variant1);
    }
    fn new_variant2() -> Self {
        Foo(FooKind::Variant2);
    }
}

struct Bar {
    foo: Foo
}
```

https://github.com/Emurgo/cardano-serialization-lib/blob/f2403651fbe76c95db5e324eb4d62d7765d82501/rust/src/protocol_types/plutus/redeemer_tag.rs#L17-L72
https://github.com/Emurgo/cardano-serialization-lib/blob/f2403651fbe76c95db5e324eb4d62d7765d82501/rust/src/protocol_types/plutus/redeemer.rs#L7-L8

Reply:
Seems the old way little bit overcomplicated for simple enum. Since this version will be pushing to use approach similar to `VoteKind`